### PR TITLE
Transact pro

### PR DESCRIPTION
**Pull request by varyonic**
New gateway for Transact Pro, www.transactpro.lv.

### CODE REVIEW

**Comment by varyonic** [comment]
Corrected test host and SSL cert.

**Comment by girasquid** [comment]
If the host + SSL cert is corrected, can you remove the `OpenSSL::SSL::VERIFY_NONE`?

**Comment by varyonic** [comment]
Removed.

File: lib/active_merchant/billing/gateways/transact_pro.rb [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+    # For more information visit {Transact Pro Services}[https://www.transactpro.lv/business/]
+    #
+    # This gateway was formerly associated with www.1stpayments.net
+    #
+    # Written by Piers Chambers (Varyonic.com)
+    class TransactProGateway < Gateway
+      self.test_url = 'https://gw2sandbox.tpro.lv:8443/gw2test/gwprocessor2.php'
+      self.live_url = 'https://www2.1stpayments.net/gwprocessor2.php'
+
+      class SslConnection < ActiveMerchant::Connection
```
Comment (j-mutter) [inline lib/active_merchant/billing/gateways/transact_pro.rb]: Uh, am I missing something here? This class seems useless, and you could just use an `ActiveMerchant::Connection` in your `new_connection` method below

File: lib/active_merchant/billing/gateways/transact_pro.rb [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+
+      def success_from(response)
+        !! (response[:status] =~ /success/i or response[:status] =~ /ok/i)
+      end
+
+      def message_from(response)
+        response[:message] || response[:status]
+      end
+
+      # def authorization_from(response)
```
Comment (j-mutter) [inline lib/active_merchant/billing/gateways/transact_pro.rb]: This can be deleted instead of commented out

**Comment by varyonic** [comment]
Removed redundant code in full.

**Comment by ntalbott** [comment]
@varyonic can you get me remote credentials so I can run those tests?

**Comment by varyonic** [comment]
Mailed you the credentials and test card details, I will need your test IP address to add to the gateway firewall.

**Comment by ntalbott** [comment]
@varyonic I got your email, and will drop you a line once I'm both at my house and have some time to work on it, to get the IP address whitelisted. Thanks!

**Comment by ntalbott** [comment]
@varyonic and finally I'm getting a chance to work on this. Just emailed you back privately. Any chance Transact Pro has changed their stance on public test credentials? That would make it a lot easier to get changes merged!

**Pull request closed without merging**